utility/service/service.go

```diff
@@ -147,3 +151,151 @@ func (r *relay) GetRelayChain() RelayChain              { return nil }
 func (r *relay) GetGeoZone() GeoZone                    { return nil }
 func (r *relay) GetToken() AAT                          { return nil }
 func (r *relay) GetSignature() string                   { return "" }
+
+type Provider interface {
+	CurrentHeight() int64
+	GetSession(appAddr string, height int64, relayChain, geoZone string) (*coreTypes.Session, error)
+}
+
+type Servicer interface {
+	HandleRelay(*coreTypes.Relay) (*coreTypes.RelayResponse, error)
+}
+
+func NewServicer(provider Provider, config configs.ServicerConfig) (Servicer, error) {
```

Comment:
We have a pattern for factory functions. See `shared/modules/factory.go` as an example. This will also give you access to the `bus` I mentioned above.

Reply:
Thank you for the pointer. Fixed.